### DISCUSSION TITLE
Presenting Multiple Stimuli

### DISCUSSION THREAD

**Discussion by mcvc968** [jspsych/jsPsych#1131] [General]
Hi all, 

I am trying to code a serial reaction time task where I present two stimuli (an image and audio) simultaneously while the participant reacts to the image. I know there is a plugin for audio-keyboard responses and one for image-keyboard but is there a way to combine the two? Or is there a way to integrate audio with the serial-reaction-time plugin?

Sorry, I am pretty new to this.

**Comment by becky-gilbert** [comment]
One thing you could do is use the audio-keyboard-response plugin to play the audio, and add the image to the screen via the `prompt` parameter, like this:
```javascript
    var start = {
        type: 'html-button-response',
        choices: ['Start'],
        stimulus: '<p>The user needs to interact with the page before the audio will play.</p>'
    };

    var trial = {
        type: 'audio-keyboard-response',
        stimulus: 'jspsych-6.1.0/examples/sound/speech_blue.mp3',
        choices: ['f','j'],
        prompt: '<img src="jspsych-6.1.0/examples/img/1.gif">' 
    };
    
    jsPsych.init({
        timeline: [start, trial],
        preload_images: ['jspsych-6.1.0/examples/img/1.gif'],
        use_webaudio: false,
        on_finish: function () {
            jsPsych.data.displayData();
        }
    });
```
Note that if you do it this way, you need to [manually preload the images](https://www.jspsych.org/overview/media-preloading/#manual-preloading) in `jsPsych.init` because they're embedded in the prompt string, rather than being passed into the plugin directly via an image parameter. Also, I used `use_webaudio: false` in `jsPsych.init` so that the audio will play when you open the HTML file directly in the browser (rather than loading it via a server).

Does that do what you need?
